sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/OnTimerMethodSpecifier.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.beam.sdk.transforms.reflect;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.transforms.DoFn;
+
+/**
+ * Used by {@link ByteBuddyDoFnInvokerFactory} to Dynamically generates
+ * {@link OnTimerInvoker} instances for invoking a particular
+ * {@link DoFn.TimerId} on a particular {@link DoFn}.
+ */
+
+@AutoValue
+abstract class OnTimerMethodSpecifier {
+    public abstract Class<? extends DoFn<?, ?>> fnClass();
+    public abstract String timerId();
+    public static OnTimerMethodSpecifier
+    create(Class<? extends DoFn<?, ?>> fnClass, String timerId){
```

Comment:
This is just minor thing, but I don't like using `create` for these kinds of constructors, because it sounds like you are always building a new object. But since all the pieces are immutable, object allocation is an implementation detail that should not be mentioned. So I like `of` or `for` or `forClassAndTimerId`. These can also be more descriptive, in case we add new constructors with different arguments.